.github/workflows/pull-request.yml

```diff
@@ -113,3 +113,17 @@ jobs:
         with:
           path: './coverage/lcov.info'
           min_coverage: 94.3
+
+  Type-Checker:
+    name: Checks Types of all variables
+    runs-on: ubuntu-latest
+    needs: Linter
+    steps: 
+      - name: Checkout repository
+        uses: actions/checkout@v3
+
+      - name: Install dependencies
+        run: npm ci  
+      
+      - name: Run Typesctipt Type-Checker
```

Comment:
1. Add this under `Linter` as it is a similar function. We don't need a separate `Type-Checker ` job section. The `Linter` section is already running `npm ci`.  
1. Also make `JSDocs` depend on `Test-Application`. It doesn't make sense for us to be automatically generating documentation before knowing whether the tests pass.